tests/Block/AuditBlockServiceTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Block;
+
+use SimpleThings\EntityAudit\AuditReader as SimpleThingsAuditReader;
+use SimpleThings\EntityAudit\Revision;
+use Sonata\BlockBundle\Block\BlockContext;
+use Sonata\BlockBundle\Model\Block;
+use Sonata\BlockBundle\Test\AbstractBlockServiceTestCase;
+use Sonata\DoctrineORMAdminBundle\Block\AuditBlockService;
+
+/**
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class AuditBlockServiceTest extends AbstractBlockServiceTestCase
+{
+    private $simpleThingsAuditReader;
+    private $blockService;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->simpleThingsAuditReader = $this->prophesize(SimpleThingsAuditReader::class);
+
+        $this->blockService = new AuditBlockService(
+            'block.service',
+            $this->templating,
+            $this->simpleThingsAuditReader->reveal()
+        );
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testExecute()
+    {
+        $block = $this->prophesize(Block::class);
+        $revision = $this->prophesize(Revision::class);
```

Comment:
This one you can easily avoid mocking

Reply:
Both, in fact.

Reply:
even if  I assert them down?

Reply:
For `$revision` it will simplify things, and for `$block`, you did not write any assertions. The only thing lost is the number of times `getRev` is called, but that's an implementation detail, and we only care about the outcome of the test, right?